compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3895,9 +3939,35 @@ private BType validateOverlappingFields(BRecordType recordType, BField origField
         return symTable.semanticError;
     }
 
+    private void removeErrorFromCompoundTypes(List<BType> remainingTypes) {
+        Iterator<BType> remainingIterator = remainingTypes.listIterator();
+        boolean needToAddAnyAndReadonly = false;
```

Comment:
Can't we just call this
```suggestion
        boolean addAnyAndReadonly = false;
```